integration-tests/utils/integration-helpers.js

```diff
@@ -72,9 +72,60 @@ const withUser = test => async args => {
   }
 }
 
+const addUserToBilling = async ({ email }) => {
+  const ajaxPage = await context.newPage()
+
+  await ajaxPage.goto(testUrl)
+  await signIntoTerra(ajaxPage)
+
+  await ajaxPage.evaluate((email, billingProject) => {
+    return window.Ajax().Billing.project(billingProject).addUser(['User'], email)
+  }, email, billingProject)
+  await ajaxPage.close()
+}
+
+const removeUserFromBilling = async ({ email }) => {
+  const ajaxPage = await context.newPage()
+
+  await ajaxPage.goto(testUrl)
+  await signIntoTerra(ajaxPage)
+
+  await ajaxPage.evaluate((email, billingProject) => {
+    return window.Ajax().Billing.project(billingProject).removeUser(['User'], email)
+  }, email, billingProject)
+  await ajaxPage.close()
+}
+
+const withBilling = test => async ({ email, ...args }) => {
+  await addUserToBilling({ email })
+
+  try {
+    await test({ ...args, email })
+  } finally {
+    await removeUserFromBilling({ email })
+  }
+}
+
+const withRegisteredUser = test => withUser(async ({ page, token, ...args }) => {
```

Comment:
For later: At mobbing, we'd talked about having `with*` functions make API calls instead of driving the UI. Then these steps would move back into the register-user test.

Reply:
I will add this to a second PR of clean up and optimizing